modules/autoscaling-policy-metric-alarm-pair/main.tf

```diff
@@ -0,0 +1,59 @@
+variable "name" {
+  description = "Prefix of the AutoScaling Policy."
+}
+
+variable "asg_name" {
+  description = "Name of the AutoScaling Group that this policy would apply to."
+}
+
+variable "metric" {
+  description = "CPUUtilization or MemoryUtilization"
+}
+
+resource "aws_autoscaling_policy" "scale_up" {
+  name                   = "${var.name}-asp-up"
+  autoscaling_group_name = "${var.asg_name}"
+  adjustment_type        = "ChangeInCapacity"
+  cooldown               = "300"
+  scaling_adjustment     = "3"
+  policy_type            = "SimpleScaling"
+}
+
+resource "aws_cloudwatch_metric_alarm" "scale_up" {
+  alarm_name          = "${var.name}-scale-up-alarm"
+  comparison_operator = "GreaterThanOrEqualToThreshold"
+  evaluation_periods  = "3"
+  metric_name         = "${var.metric}"
+  namespace           = "AWS/EC2"
+  period              = "60"
+  statistic           = "Average"
+  threshold           = "60"
+  dimensions {
+    AutoScalingGroupName = "${var.asg_name}"
+  }
+  alarm_actions = ["${aws_autoscaling_policy.scale_up.arn}"]
+}
+
+resource "aws_autoscaling_policy" "scale_down" {
+  name                   = "${var.name}-asp-down"
```

Comment:
`asg` in this and the `scale_up`

Reply:
Oh, I did not mean "asg", I mean "AutoScaling Policy".